code/datums/elements/surgery_initiator.dm

```diff
@@ -0,0 +1,147 @@
+/// Allows an item to  be used to initiate surgeries.
+/datum/element/surgery_initiator
+	element_flags = ELEMENT_DETACH
+
+/datum/element/surgery_initiator/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, .proc/initiate_surgery_moment)
+
+/datum/element/surgery_initiator/Detach(datum/source)
+	UnregisterSignal(source, COMSIG_ITEM_ATTACK)
+	return ..()
+
+/// Does the surgery initiation.
+/datum/element/surgery_initiator/proc/initiate_surgery_moment(datum/source, atom/target, mob/user)
+	SIGNAL_HANDLER
+	if(!isliving(target))
+		return
+	INVOKE_ASYNC(src, .proc/do_initiate_surgery_moment, source, target, user)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/datum/element/surgery_initiator/proc/do_initiate_surgery_moment(datum/source, atom/target, mob/user)
+	var/mob/living/livingtarget = target
+	var/mob/living/carbon/carbontarget
+	var/obj/item/bodypart/affecting
+	var/selected_zone = user.zone_selected
+	if(iscarbon(livingtarget))
+		carbontarget = livingtarget
+		affecting = carbontarget.get_bodypart(check_zone(selected_zone))
+
+	var/datum/surgery/current_surgery
+
+	for(var/i_one in livingtarget.surgeries)
```

Comment:
That's not a really descriptive variable name. Do something like `for(var/surgery in living_target.surgeries)`